kubemarine/procedures/check_iaas.py

```diff
@@ -314,12 +315,158 @@ def check_access_to_thirdparties(cluster: KubernetesCluster):
+        local_path = utils.get_resource_absolute_path("resources/scripts/check_url_availability.py",
+                                                  script_relative=True)
+        random_temp_path = "/tmp/%s.py" % uuid.uuid4().hex
+        all_group.put(local_path, random_temp_path, binary=False)
+
+        for node in all_group.get_ordered_members_list(provide_node_configs=True):
+            # Check if resolv.conf is actual
+            resolv_conf_actual = check_resolv_conf(node['connection'])
+
+            if not resolv_conf_actual:
+                warnings.append(f"resolv.conf is not installed for node {node['connect_to']}: "
+                                f"Package repositories can be unavailable. You can install resolv.conf using task "
+                                f"`install --tasks prepare.dns.resolv_conf`")
+
+            # Check with script
+            python_executable = cluster.context['nodes'][node['connect_to']]['python']['executable']
+            for repository_url in repository_urls:
+                res = node['connection'].run("%s %s %s %s" % (python_executable, random_temp_path, repository_url,
+                                                          cluster.inventory['timeout_download']), warn=True)
+                _, result = list(res.items())[0]
+                if result.failed and resolv_conf_actual:
+                    broken.append(f"{node['connect_to']}, {repository_url}: {result.stderr}")
+                elif result.failed:
+                    warnings.append(f"{node['connect_to']}, {repository_url}: {result.stderr}")
+
+        # Remove file
+        rm_command = "rm %s" % random_temp_path
+        all_group.run(rm_command)
+
+        if broken:
+            raise TestFailure('Found problems for package repositories', hint=yaml.safe_dump(broken))
+        elif warnings:
+            raise TestWarn('Found potential problems for package repositories', hint=yaml.safe_dump(warnings))
+        tc.success('All package repositories are correct and available')
+
+
+def check_access_to_packages(cluster: KubernetesCluster):
+    broken = []
+    warnings = []
+
+    for node in cluster.nodes['all'].get_ordered_members_list(provide_node_configs=True):
+        # Check if package repos are actual
+        package_repos_are_actual = check_package_repositories(node['connection'])
+        if not package_repos_are_actual:
+            warnings.append(f"Package repositories are not installed for {node['connect_to']}: "
+                            f"Packages can be unavailable. You can install it using tasks "
+                            f"`install --tasks prepare.dns.resolv_conf,prepare.package_manager.configure`")
+
+        packages_to_check = cluster.inventory["services"]["packages"].get("install", {}).get('include', []).copy()
+        for association_name, association in cluster.inventory["services"]["packages"].get("associations", {})\
+                .get(node['connection'].get_nodes_os(), {}).items():
```

Comment:
Not all associations are in general required. For example, audit is not managed by Kubemarine for rhel OS family (though, probably still need to check it), haproxy/keepalived are not required for some non-HA schemes.

Reply:
Added package associations filtration for nodes, where they should be installed